openlcs/libs/sc_handler.py

```diff
@@ -0,0 +1,130 @@
+import koji
+import glob
+import os
+import uuid
+import sys
+
+# Fix absolute import issue in openlcs.
+openlcs_dir = os.path.abspath(os.path.join(os.path.dirname(__file__), '..'))
+if openlcs_dir not in sys.path:
+    sys.path.append(openlcs_dir)
+from libs.common import compress_source_to_tarball  # noqa: E402
+
+
+class SourceContainerHandler(object):
+    """
+    Object used for handle source in source containers.
+    """
+    def __init__(self, config=None):
+        self.config = config
+
+    @staticmethod
+    def get_component_flat(data, comp_type):
+        return {
+            'uuid': str(uuid.uuid4()),
+            'type': comp_type,
+            'name': data.get('name'),
+            'version': data.get('version'),
+            'release': data.get('release'),
+            'license': '',
+            'arch': '',
```

Comment:
for srpm, the arch should be "src", not  ""

Reply:
Added one line component['arch'] = 'src' for get_srpm_components function.

Reply:
Add the arch via koji.Parse(nvra).